ipn/ipnlocal/serve.go

```diff
@@ -257,6 +258,111 @@ func (b *LocalBackend) ServeConfig() ipn.ServeConfigView {
 	return b.serveConfig
 }
 
+// StreamFunnel opens a stream to write any incoming connections made
+// to the given HostPort out to the listening io.Writer.
+//
+// If Funnel was not already enabled for the HostPort, it is enabled
+// in the ServeConfig for the duration of the context's lifespan and
+// then turned off once the context is done.
+// If Funnel was already enabled, the ServeConfig is untouched.
+func (b *LocalBackend) StreamFunnel(ctx context.Context, w io.Writer, hp ipn.HostPort) (err error) {
```

Comment:
I wonder if we should associate this with Funnel vs. Serve or something else. WDYT?